htdocs/web_portal/controllers/site/edit_api_auth.php

```diff
@@ -47,48 +70,85 @@ function edit_entity() {
 
     // Validate the user has permission to edit properties
     if (!$serv->userCanEditSite($user, $site)) {
-        throw new \Exception("Permission denied: a site role is required to edit authentication entities at " . $site->getShortName());
+        throw new \Exception(
+            "Permission denied: A site role is required " .
+            "to edit authentication entities at " .
+            $site->getShortName()
+        );
     }
 
-    if($_POST) {     // If we receive a POST request it's to edit an authentication entity
-        submit($user, $authEnt, $site, $serv);
-    } else { // If there is no post data, draw the edit authentication entity form
-        draw($user, $authEnt, $site);
-    }
+    return [$user, $authEnt, $site, $serv];
 }
 
-function draw(\User $user = null, \APIAuthentication $authEnt = null, \Site $site = null) {
-    if(is_null($user)){
-        throw new Exception("Unregistered users can't edit authentication credentials");
+/**
+ * Helper to draw either the edit or renewal authentication entity form.
+ *
+ * @param \User|null $user
+ * @param \APIAuthentication|null $authEntity
+ * @param \Site|null $site
+ * @throws \Exception
+ */
+function draw(
+    \User $user = null,
+    \APIAuthentication $authEnt = null,
+    \Site $site = null
+) {
+    if (is_null($user)) {
+        throw new Exception(
+            "Unregistered users can't edit authentication credentials."
+        );
     }
 
     $params = array();
     $params['site'] = $site;
     $params['authEnt'] = $authEnt;
     $params['authTypes'] = array();
-    $params['authTypes'][]='X.509';
-    $params['authTypes'][]='OIDC Subject';
+    $params['authTypes'][] = 'X.509';
+    $params['authTypes'][] = 'OIDC Subject';
     $params['user'] = $user;
 
+    if ($_REQUEST['isRenewalRequest']) {
+        $params['isRenewalRequest'] = true;
+    }
+
     show_view("site/edit_api_auth.php", $params);
     die();
 }
 
-function submit(\User $user, \APIAuthentication $authEnt, \Site $site, org\gocdb\services\Site $serv) {
-    $newValues = getAPIAuthenticationFromWeb();
+/**
+ * If this receives a POST request,
+ * it can be either to edit an API authentication entity or
+ * to update the `$lastRenewTime` in `APIAuthentication`.
+ *
+ * @param \User $user
+ * @param \APIAuthentication $authEntity
+ * @param \Site $site
+ * @param org\gocdb\services\Site $service
+ */
+function submit(
+    \User $user,
+    \APIAuthentication $authEnt,
+    \Site $site,
+    org\gocdb\services\Site $serv
+) {
+    $params = array();
+
+    if ($_REQUEST['isRenewalRequest']) {
+        $newValues['isRenewalRequest'] = $params['isRenewalRequest'] = true;
```

Comment:
```suggestion
        $newValues['isRenewalRequest'] = true;
        $params['isRenewalRequest'] = true;
```

if that is indeed the desired behaviour, it's clearer what's going on and doesn't have two assignments on one line.

Reply:
It is indented behaviour. Sure, I have removed assigning multiple variable at once.